woocommerce/woocommerce-ios
# HACK week: Remove WPKit from Networking and UI layers

Part 3/3 of the HACK week project to remove WordPressKit as a dependency in the app.


## Description

The previous 2 PRs removed WordPressKit from the Yosemite layer, so this one takes care of the rest by removing the dependency from the Networking and UI layers.

For the Networking layer, we only need `WordPressShared` in `ApplicationPasswordStorage` to use the `Secret` type, so `WordPressKit` has been replaced by this dependency.

For the UI layer, `WordPressKit` was used in `AuthenticationManager` to check for error codes. The error handling has been updated to use just the name of the error code, and the XMLRPC error check has been removed since XMLRPC is no longer a requirement for the app (removed in #8790 and #8259).

## Testing instructions

The app should still build successfully and CI should pass. In podfile there should be no mention of `WordPressKit` anymore, we only have the dependency as required in `WordPressAuthenticator`.

**Testing application password**:
- Log out of the app and log in to a site without Jetpack.
- Notice that all features still work correctly after login.

**Testing login error handling**: 
- Log out of the app.
- Select Enter your site address and enter the address of a Jetpack site.
- Enter the email address that doesn't belong to any WPCom account.
- Notice that an error screen is displayed for unknown user.

<img src="https://user-images.githubusercontent.com/5533851/223606220-6479b86d-f3a6-4ae3-b000-6cbd955d6cb8.png" width=320 />

 
## Screenshots

N/A

---
